pylearn2/linear/cudnn2d.py

```diff
@@ -0,0 +1,203 @@
+import functools
+import numpy as np
+
+from theano.sandbox.cuda.dnn import GpuDnnConv, GpuDnnConvDesc
+from theano.sandbox.cuda.basic_ops import gpu_contiguous
+
+from pylearn2.packaged_dependencies.theano_linear.conv2d \
+    import Conv2d as OrigConv2D
+
+from pylearn2.linear.linear_transform import LinearTransform as P2LT
+from pylearn2.utils import sharedX
+from pylearn2.utils.rng import make_np_rng
+
+
+default_seed = [2012, 11, 6, 9]
+default_sparse_seed = [2012, 11, 6]
+
+
+class Cudnn2D(OrigConv2D):
+    """
+    Wrapper on the Theano Cudnn op.
+
+    Parameters
+    ----------
+    filters : Theano shared variable
+        4D-tensor of shape (out channels, in channels, rows, cols)
+    batch_size : int
+        The size of the input batches
+    input_space : Space
+        The Space of the input data
+    output_axes : tuple, optional
+        The requested output axes. If not specified `bc01` will be used.
```

Comment:
Shuffle should be performed in place by numpy. Actually if I am not wrong, it should not even modify the array in memory, but directly index it in the right way when you access the data. 
Anyway, if you still think it is better to default to bc01 I'm ok with that.